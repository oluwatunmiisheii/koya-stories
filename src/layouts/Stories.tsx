import React from 'react';

interface IStoriesLayoutProps {
  children: React.ReactNode; 
}
 
const StoriesLayout: React.FC<IStoriesLayoutProps> = ({ children }) => {
  return ( 
    <div className="container mx-auto px-4 sm:px-6 lg:px-8 mt-16">
      {children}
    </div>
  );
}
 
export default StoriesLayout;