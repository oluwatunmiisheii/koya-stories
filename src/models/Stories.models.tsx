export interface IStory {
  id: number;
  status: string;
  title: {
    rendered: string;
  },
  excerpt: {
    rendered: string;
  },
  jetpack_featured_media_url: string;
  content?: {
    rendered: string;
  }
  date_gmt?: string;
}