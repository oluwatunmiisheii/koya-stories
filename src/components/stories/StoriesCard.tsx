import * as React from 'react';
import { IStory } from '../../models/Stories.models';
import { truncate } from '../../util/TruncateText';
import { Link } from 'react-router-dom'
 
interface IStoriesCardProps {
  story: IStory;
}

const StoriesCard: React.FC<IStoriesCardProps> = ({story}) => {
  return (  
    <div className="flex flex-col bg-white shadow-sm rounded border border-gray-200 relative pb-12">
      <div className="flex-shrink-0">
        <img className="h-64 w-full object-cover" src={story.jetpack_featured_media_url} alt="avatar" />
      </div>
      <div className='mt-3 p-5 relative'>
        <Link to={`stories/${story.id}`} className="text-lg font-bold cursor-pointer underline" dangerouslySetInnerHTML={{__html: truncate(story.title.rendered, 200)}}></Link>
        <div className="mt-1 mb-5" dangerouslySetInnerHTML={{__html: truncate(story.excerpt.rendered, 200)}}></div>
      </div>
      <Link
        to={`stories/${story.id}`}
        className="absolute bottom-0 mb-6 ml-5 inline-flex items-center px-4 py-2 border border-transparent text-base font-medium rounded-md shadow-sm text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
      >
        See More
      </Link>
    </div>
  );
}
 
export default StoriesCard;