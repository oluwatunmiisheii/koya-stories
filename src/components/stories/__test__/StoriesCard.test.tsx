import React from 'react';
import { render, screen } from '@testing-library/react';
import StoriesCard from '../StoriesCard';
import { IStory } from '../../../models/Stories.models';
import { MemoryRouter as Router } from 'react-router-dom';

const storiesMock: IStory = {
  id: 1,
  status: 'publish',
  title: {
    rendered: 'Test Story'
  },
  excerpt: {
    rendered: 'Test Excerpt'
  },
  jetpack_featured_media_url: 'https://via.placeholder.com/150'
}

const setup = () => render(
  <Router>
    <StoriesCard story={storiesMock} />
  </Router>
);
 

describe('StoriesCard', () => {
  it('should render title', async () => {
    const { container } = setup();
  
    const title = screen.getByText(/^Test story$/i); // full string match, ignore case
  
    expect(title).toBeInTheDocument();
    expect(title).toHaveTextContent('Test Story');
  
    expect(container).toMatchSnapshot();
  });
})
