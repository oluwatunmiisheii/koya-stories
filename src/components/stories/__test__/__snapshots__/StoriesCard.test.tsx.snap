// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoriesCard should render title 1`] = `
<div>
  <div
    class="flex flex-col bg-white shadow-sm rounded border border-gray-200 relative pb-12"
  >
    <div
      class="flex-shrink-0"
    >
      <img
        alt="avatar"
        class="h-64 w-full object-cover"
        src="https://via.placeholder.com/150"
      />
    </div>
    <div
      class="mt-3 p-5 relative"
    >
      <a
        class="text-lg font-bold cursor-pointer underline"
        href="/stories/1"
      >
        Test Story
      </a>
      <div
        class="mt-1 mb-5"
      >
        Test Excerpt
      </div>
    </div>
    <a
      class="absolute bottom-0 mb-6 ml-5 inline-flex items-center px-4 py-2 border border-transparent text-base font-medium rounded-md shadow-sm text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
      href="/stories/1"
    >
      See More
    </a>
  </div>
</div>
`;
